### ISSUE TITLE
Add a bson tag in structures

### ISSUE THREAD

**Issue by RegularITCat** [freetaxii/libstix2#14]
I would like to use golang mongodb driver with this library, but it seems mongodb wont decode automatically to structure until bson tag in structure is defined.
`Pattern        string   'json:"pattern,omitempty"'`
change to something like this
`Pattern        string   'json:"pattern,omitempty" bson:"pattern"'`

**Comment by DallasP9124** [comment]
@RegularITCat this repo is essentially dead. It has not been updated in years therefor not compliant with the stix 2.1 official spec.

**Comment by RegularITCat** [comment]
So... Then we should ressurect that, yea?
I'm not very familiar with stix specs, but already added a bson tags in my repository, github.com/RegularITCat/libstix2
I'm really appreciate if you add issues in this and help me understand difference between current version of library and stix2.1

**Comment by DallasP9124** [comment]
Well ya if you want to fork and update it sure go ahead, that's the best way to bring old repos back to life. I frankly don't have the time or interest to bring this code up to date with the official spec, but, if that is something worthwhile to you then I would review the [official 2.1 spec](https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html) and begin writing out issues in your fork and managing it there.

If you happen to get your fork updated and spec compliment, you can do a pull request from your fork.

However, because the repo is dead and little has been done by the owners in the past years, it's not likely the PR will be accepted.

Best of luck 🤞

**Comment by jordan2175** [comment]
You are correct that I have not pushed an updated in some time. I have been a bit busy with other work. I did do some work early on to bring the library current with 2.1. And yes, I do know what is in the 2.1 spec since I used to Chair that work and was the leader Editor for the spec. There are a few things that are not in there though and I admit that.  I generally accept most pull requests as long as they are not do outrageous.
